provider/awssd/aws_sd.go

```diff
@@ -164,28 +164,29 @@ func (p *AWSSDProvider) Records(ctx context.Context) (endpoints []*endpoint.Endp
 		}
 
 		for _, srv := range services {
-			instances, err := p.ListInstancesByServiceID(srv.Id)
+			resp, err := p.client.DiscoverInstancesWithContext(ctx, &sd.DiscoverInstancesInput{
+				NamespaceName: ns.Name,
+				ServiceName:   srv.Name,
+			})
 			if err != nil {
 				return nil, err
 			}
 
-			if len(instances) > 0 {
-				ep := p.instancesToEndpoint(ns, srv, instances)
-				endpoints = append(endpoints, ep)
-			}
-			if len(instances) == 0 {
-				err = p.DeleteService(srv)
-				if err != nil {
-					log.Warnf("Failed to delete service \"%s\", error: %s", aws.StringValue(srv.Name), err)
+			if len(resp.Instances) == 0 {
+				if err := p.DeleteService(srv); err != nil {
+					log.Warnf("Failed to delete service %q, error: %s", aws.StringValue(srv.Name), err)
```

Comment:
no warnings if it's not a deprecation.
Please change to Errorf or Infof

Reply:
updated
we got some Wanrf invocations left there, I could replace them here, or should I cut separate PR or there are some other plans on them?